content/blog/gatsby-multi-author/index.md

```diff
@@ -0,0 +1,231 @@
+
+  //...
+}
+```
+
+### Bouns: Author profile picture
+
+If you look at the previous snippets you will notice we used the `profilepicture` to
+support author's profile pictures. Notice how in the GraphQL query before we used the
+`childImageSharp` query to enable image optimization, something that the Gatbsy Starter blog
+does by default in the mono-author mode.
+
+By using this `childImageSharp` and any of the other queries of `gatsby-image` you can
+transform dynamically loaded images via the expected grapqhl interface wiht ease.
+
+## How to use all authors
+
+Right now we only cover the pseudo magical way of mapping a Front Matter
+attribute to a filesystem database i.e. `author.yaml`.
+
+But what happens when you want all the authors, lets say for a list of
+authors in your page or a shared about page?
+
+Gatsby also has you covered with this by using the `filesystem` plugin
+that we setup before Gatsby will automaticllly provide a way of
+reading the entire `author.yaml`:
+
+```javascript:title=src/pages/about.js
+export const pageQuery = graphql`
+    allAuthorYaml {
+      nodes {
+        id
+        bio
+      }
+    }
+  }
+`
+```
+
+Notice how `author.yaml` is mapped to `allAuthorYaml`, that name transformation
+is significant and you should be carefuly with it.
+
+If unsure you can always go to the graphql explored that Gatsby provides by
+default and play around with the available queries.
+
+At this point your page component will be provided with an array of authors:
+
+```javascript:title=src/pages/about.js
+export default function About(props) {
+  // this is an array of authors
+  const authors = props.allAuthorYaml.nodes
+
+  //...
+}
+```
+
+### Bonus: Run-Time multi author support
+
+Notice that is also enables you to bypass the Graphql mapping interface and
+do an ad-hoc manual implementation of the muli author support.
+I will probably not recommend it for this use case but it is a posibility if you
+encounter a use case that might benefit from it. Notice that by using this you dont need
+to statically know the author id like we did in the `blog-post.js` before, you can
+do it entirely dynamically by simply passing react props.
+
+```javascript:title=src/components/MyComponents.js
+export default function Author(props) {
+  const authorId = props.author
+  const authors = props.allAuthorYaml.nodes
+
+  // manually search for the selected author
+  const myAuthor = authors.find(a => a.id === authorId)
+
+  return <p>Author Bio: {myAuthor.bio}</p>
+}
+
+export const query = graphql`
+    allAuthorYaml {
+      nodes {
+        id
+        bio
+      }
+    }
+  }
+`
+```
+
+And you can use it like:
+
+```javascript
+<Author author="author1" />
+```
+
+This can be extrapolated to a lot of other use cases.
+
+## Closing
+
+I hope that I have eased the entry barrier for a multi author blog in Gatbsy.js.
+
+This blog is actually a real life implementation of the multi author blog. Check our [source][3]!
```

Comment:
I think this could be commented on the first sections of the post to make it more relatable (and so they can see how the solution looks from the beginning)?

Reply:
done